includes/class-sensei-course.php

```diff
@@ -100,8 +100,9 @@ public function __construct () {
         add_action('save_post', array( 'Sensei_Course', 'flush_rewrite_rules' ) );
 
 		// Allow course archive to be setup as the home page
-		add_action( 'pre_get_posts', array( $this, 'allow_course_archive_on_front_page' ) );
-
+		if ( (int) get_option( 'page_on_front' ) > 0 ) {
```

Comment:
...it also basically carries over this existing piece of logic:

`0 == intval( get_option( 'page_on_front' ) )`

It just applies the test earlier on (and avoids adding a callback altogether if it isn't satisfied).